draft-ietf-tls-external-psk-importer.md

```diff
@@ -79,156 +80,151 @@ when, and only when, they appear in all capitals, as shown here.
 
 # Overview
 
-Intuitively, key importers mirror the concept of key exporters in TLS in that they
-diversify a key based on some contextual information before use in a connection. In contrast to
-key exporters, wherein differentiation is done via an explicit label and context string,
-the key importer defined herein uses a label and set of hash algorithms to
-differentiate an external PSK into one or more PSKs for use.
+Key importers mirror the concept of key exporters in TLS in that they diversify a key
+based on some contextual information before use in a connection. In contrast to key exporters,
+wherein differentiation is done via an explicit label and context string, the key importer
+defined herein uses an optional context string along with a target protocol and KDF
+identifier to differentiate an external PSK into one or more PSKs for use.
 
 Imported keys do not require negotiation for use, as a client and server will not agree upon
 identities if not imported correctly. Thus, importers induce no protocol changes with
-the exception of expanding the set of PSK identities sent on the wire. Endpoints may incrementally
-deploy PSK importer support by offering non-imported keys for TLS versions prior to TLS 1.3.
-(Negotiation and use of imported PSKs requires both endpoints support the importer API described herein.)
+the exception of expanding the set of PSK identities sent on the wire. Endpoints may
+incrementally deploy PSK importer support by offering non-imported keys for TLS versions
+prior to TLS 1.3. (Negotiation and use of imported PSKs requires both endpoints support
+the importer API described herein.)
+
+Clients which import external keys TLS MUST NOT use these keys for any other purpose.
+Moreover, each external PSK MUST be associated with at most one hash function.
 
 ## Terminology {#terminology}
 
-- External PSK (EPSK): A PSK established or provisioned out-of-band, i.e., not from a TLS connection, which is
-a tuple of (Base Key, External Identity, KDF). The associated KDF (and hash function) may be undefined.
+- External PSK (EPSK): A PSK established or provisioned out-of-band, i.e., not from a TLS
+connection, which is a tuple of (Base Key, External Identity, Hash). The associated hash
+function may be undefined.
```

Comment:
I think that you should drop the "hash is undefined" piece.  There is no reason to have that.  These are for exclusive use and provision of the key can just assume a default hash function if they want to avoid the overhead of having to say "SHA-256" every time.

Reply:
That seems reasonable to me. Dropped!